lightning/src/ln/peer_handler.rs

```diff
@@ -819,7 +814,9 @@ impl<Descriptor: SocketDescriptor, CM: Deref> PeerManager<Descriptor, CM> where
 						let (mut descriptor, peer) = get_peer_for_forwarding!(node_id, {
 								//TODO: Drop the pending channel? (or just let it timeout, but that sucks)
 							});
-						peer.pending_outbound_buffer.push_back(peer.channel_encryptor.encrypt_message(&encode_msg!(msg)));
+						if let PeerState::Connected(ref mut conduit) = peer.encryptor {
+							peer.pending_outbound_buffer.push_back(conduit.encrypt(&encode_msg!(msg)));
+						}
```

Comment:
If we have a peer in `node_id_to_descriptor`, it should also have a conduit. Any chance this can be done within `get_peer_for_forwarding` so the repetition across the method could be removed? We probably want to assert this as well.

Reply:
Absolutely, though here too, I think that it will be much clearer once get_peer_for_forwarding is refactored from being a macro, which I strongly believe it doesn't need to be.

Reply:
I tend to agree the redundancy of all these calls suck. Happy for y'all to hash out which PR it gets fixed in, though.